community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -757,9 +765,12 @@ private void close( Header.Writer headerWriter ) throws IOException
 
         try
         {
-            // Perform a checkpoint before closing. If no changes has happened since last checkpoint,
-            // no new checkpoint will be created.
-            checkpoint( IOLimiter.unlimited(), headerWriter );
+            if ( health.isHealthy() )
```

Comment:
@burqen @MishaDemianenko I can't agree with that. May I say that Monitor is _always_ preferred over direct Log. A Monitor has way more usability, is customizable and is easier to test. We don't use direct logging anywhere else in index component, why here?.